flytekit/control_plane/workflow.py

```diff
@@ -0,0 +1,138 @@
+                if upstream.id is None:
+                    raise _user_exceptions.FlyteAssertion(
+                        "Some nodes contained in the workflow were not found in the workflow description.  Please "
+                        "ensure all nodes are either assigned to attributes within the class or an element in a "
+                        "list, dict, or tuple which is stored as an attribute in the class."
+                    )
+        super(FlyteWorkflow, self).__init__(
+            id=id,
+            metadata=metadata,
+            metadata_defaults=metadata_defaults,
+            interface=interface,
+            nodes=nodes,
+            outputs=output_bindings,
+        )
+        self._sdk_nodes = nodes
+        self._has_registered = False
+
+    @property
+    def upstream_entities(self):
+        return set(n.executable_sdk_object for n in self._sdk_nodes)
+
+    @property
+    def interface(self) -> _interfaces.TypedInterface:
+        return super(FlyteWorkflow, self).interface
+
+    @property
+    def entity_type_text(self) -> str:
+        return "Workflow"
+
+    @property
+    def resource_type(self):
+        return _identifier_model.ResourceType.WORKFLOW
+
+    def get_sub_workflows(self) -> List["FlyteWorkflow"]:
+        result = []
+        for node in self.nodes:
+            if node.workflow_node is not None and node.workflow_node.sub_workflow_ref is not None:
+                if node.executable_sdk_object is not None and node.executable_sdk_object.entity_type_text == "Workflow":
+                    result.append(node.executable_sdk_object)
+                    result.extend(node.executable_sdk_object.get_sub_workflows())
+                else:
+                    raise _system_exceptions.FlyteSystemException(
+                        "workflow node with subworkflow found but bad executable "
+                        "object {}".format(node.executable_sdk_object)
+                    )
+
+            # get subworkflows in conditional branches
+            if node.branch_node is not None:
+                if_else: _workflow_models.IfElseBlock = node.branch_node.if_else
+                leaf_nodes: List[_nodes.SdkNode] = filter(
+                    None,
+                    [
+                        if_else.case.then_node,
+                        *([] if if_else.other is None else [x.then_node for x in if_else.other]),
+                        if_else.else_node,
+                    ],
+                )
+                for leaf_node in leaf_nodes:
+                    exec_sdk_obj = leaf_node.executable_sdk_object
+                    if exec_sdk_obj is not None and exec_sdk_obj.entity_type_text == "Workflow":
+                        result.append(exec_sdk_obj)
+                        result.extend(exec_sdk_obj.get_sub_workflows())
+
+        return result
+
+    @classmethod
+    @_exception_scopes.system_entry_point
+    def fetch(cls, project, domain, name, version=None):
```

Comment:
let's make None required.  users of this code should not have to have internal_config set up except for the endpoint/auth stuff.